service/command.go

```diff
@@ -33,6 +32,11 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 			if err := featuregate.GetRegistry().Apply(gatesList); err != nil {
 				return err
 			}
+			if BuildFlag {
+				err, _ := getBuildInfo(set)
+				return err
```

Comment:
If this is going to return and not execute the collector command then should it be a separate `cobra.Command` and invoked like this:

```sh
$ ./otelcol build-info
```

Reply:
Oh okay, thanks a lot for reviewing my code. I actually followed the pattern for the already existing --config flag and so it is called like this:

`./otelcol --build-info`
The requirement on the issue that I took up was actually to create a flag. It does not execute the collector command as far as I have checked. 
But maybe a command is now more acceptable?

Reply:
I agree with @Aneurysm9 that a command is more suitable here: the "--config" flag is a parameter that is relevant for the "run" command (implicit), while "build-info" is an action to generate this output.

Reply:
Okay, I am afraid this might introduce a breaking change. I would submit the PR